rclcpp/include/rclcpp/node_options.hpp

```diff
@@ -408,6 +409,16 @@ class NodeOptions
   NodeOptions &
   allocator(rcl_allocator_t allocator);
 
+  /// Return the time source to be used.
+  RCLCPP_PUBLIC
+  const rclcpp::node_interfaces::NodeTimeSourceInterface::SharedPtr
+  time_source() const;
+
+  /// Set the time source to be used.
```

Comment:
```suggestion
  /// Set the time source to be used.
  /**
   * This will cause the node to use the provided TimeSource,
   * usually obtained by another node. User must understand that
   * either the node that owns the TimeSource has to be spinned,
   * or `use_clock_thread_` must be configured as `true` on that node.
   */
```